cycode/cli/printers/table_printer.py

```diff
@@ -1,170 +1,168 @@
-from collections import defaultdict
-from typing import List, Dict
+from typing import List, NamedTuple
 
 import click
 from texttable import Texttable
 
-from cycode.cli.consts import LICENSE_COMPLIANCE_POLICY_ID, PACKAGE_VULNERABILITY_POLICY_ID
-from cycode.cli.models import DocumentDetections, Detection, CliError, CliResult
+from cycode.cli.printers.text_printer import TextPrinter
+from cycode.cli.utils.string_utils import obfuscate_text, get_position_in_line
+from cycode.cli.consts import SECRET_SCAN_TYPE, SAST_SCAN_TYPE, INFRA_CONFIGURATION_SCAN_TYPE
+from cycode.cli.models import DocumentDetections, Detection, CliError, CliResult, Document
 from cycode.cli.printers.base_printer import BasePrinter
 
-SEVERITY_COLUMN = 'Severity'
-LICENSE_COLUMN = 'License'
-UPGRADE_COLUMN = 'Upgrade'
-REPOSITORY_COLUMN = 'Repository'
-CVE_COLUMN = 'CVE'
-
-PREVIEW_DETECTIONS_COMMON_HEADERS = [
-    'File Path',
-    'Ecosystem',
-    'Dependency Name',
-    'Direct Dependency',
-    'Development Dependency'
+
+class ColumnInfo(NamedTuple):
+    name: str
+    width_secret: int = 1
```

Comment:
not sure I understand whats the purpose of this class

Reply:
To manage different widths of columns depends on scan type. For example in secret scan we display pretty short issue type. For example regex. But in sast we describe the issue. That's why I want to control columns widths for different scan types. This class allows to manage it